homeassistant/components/withings/__init__.py

```diff
@@ -39,18 +39,17 @@
 
 CONFIG_SCHEMA = vol.Schema(
     {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_CLIENT_ID): vol.All(cv.string, vol.Length(min=1)),
-                vol.Required(CONF_CLIENT_SECRET): vol.All(cv.string, vol.Length(min=1)),
-                vol.Optional(const.CONF_USE_WEBHOOK, default=False): cv.boolean,
-                vol.Required(const.CONF_PROFILES): vol.All(
-                    cv.ensure_list,
-                    vol.Unique(),
-                    vol.Length(min=1),
-                    [vol.All(cv.string, vol.Length(min=1))],
-                ),
-            }
+        DOMAIN: vol.All(
+            cv.deprecated(const.CONF_PROFILES, invalidation_version="0.114"),
+            vol.Schema(
+                {
+                    vol.Required(CONF_CLIENT_ID): vol.All(cv.string, vol.Length(min=1)),
+                    vol.Required(CONF_CLIENT_SECRET): vol.All(
+                        cv.string, vol.Length(min=1)
+                    ),
+                    vol.Optional(const.CONF_USE_WEBHOOK, default=False): cv.boolean,
+                }
```

Comment:
The `CONF_PROFILES` item needs to still be part of the dict schema, as optional, to not make it invalid when that item is included.

Reply:
I pushed something different, let me know if this is correct. I'm not entirely sure how the deprecated option values work. HASS contains many examples with various usage but it's still unclear to me.